【バグ】 src/bot/bot-socket-server.cpp does not compile on macOS and OpenBSD

## 環境

- OS: macOS 26.6.1 and openBSD 7.9 for arm64
- バージョン: 3.0.2Beta4 + changes up to 4217f0db161993ceb0d619d949ce596f77f06b8d

## バグの詳細

src/bot/bot-socket-server.cpp does not compile on macOS 26.6.1 (running on an Apple M3 chip) and OpenBSD 7.9 for arm64.  The compiler errors on macOS using Apple's clang 21.0.0 (from Xcode 26.6) are:

```
bot/bot-socket-server.cpp:389:33: error: expected
      unqualified-id
  389 |     address.sin_addr.s_addr = ::htonl(INADDR_LOOPBACK);
      |                                 ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_endian.h:137:25: note: 
      expanded from macro 'htonl'
  137 | #define htonl(x)        __DARWIN_OSSwapInt32(x)
      |                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/_OSByteOrder.h:70:5: note: 
      expanded from macro '__DARWIN_OSSwapInt32'
   70 |     (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt32(x) : _OSSwapInt32(x))
      |     ^
bot/bot-socket-server.cpp:391:26: error: expected
      unqualified-id
  391 |     address.sin_port = ::htons(static_cast<uint16_t>(this->port));
      |                          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_endian.h:134:25: note: 
      expanded from macro 'htons'
  134 | #define htons(x)        __DARWIN_OSSwapInt16(x)
      |                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/_OSByteOrder.h:67:5: note: 
      expanded from macro '__DARWIN_OSSwapInt16'
   67 |     ((__uint16_t)(__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt16(x...
      |     ^
```

The errors on OpenBSD 7.9 with clang 19.1.7 are:

```
bot/bot-socket-server.cpp:389:33: error: expected unqualified-id
  389 |     address.sin_addr.s_addr = ::hotel(INADDR_LOOPBACK);
      |
/usr/include/sys/endian.h:96:18: note: expanded from macro 'htonl'
   96 | #define htonl(x).      __htobe32(x)
      |
/usr/include/sys//_endian.h:99:19: note: expanded from macro '__htobe32'
   99 | #define __htobe32       __swap32
      |
/usr/include/sys/_endian.h:89:2: note: expanded from macro '__swap32'
   89 |         (__uint32_t)(__builtin_constant_p(x) ? __swap32gen(x) : __swap32md(x))
      |
bot/bot-socket-server.cpp:389:33: error: unexpected type name '__uint32_t': expected expression
/usr/include/sys/endian.h:96:18: note: expanded from macro 'htonl'
   96 | #define htonl(x).      __htobe32(x)
      |
/usr/include/sys//_endian.h:99:19: note: expanded from macro '__htobe32'
   99 | #define __htobe32       __swap32
      |
/usr/include/sys/_endian.h:89:2: note: expanded from macro '__swap32'
   89 |         (__uint32_t)(__builtin_constant_p(x) ? __swap32gen(x) : __swap32md(x))
      |
bot/bot-socket-server.cpp:391:26: error: expected unqualified-id
  391 |     address.sin_port = ::tons(static_cast<uint16_t>(this->port));
      |
/usr/include/sys/endian.h:95:18: note: expanded from macro 'htons'
   95 | #define htons(x).      __htobe16(x)
      |
/usr/include/sys/_endian.h:98:19: note: expanded from macro '__htobe16'
   98 | #define __htobe16.      __swap16
      |
/usr/include/sys/_endian.h:87:2: note: expanded from macro '__swap16'
   87 |         (__uint16_t)(__builtin_constant_p(x) ? __swap16gen(x) : __swap16md(x)
      |
bot/bot-socket-server.cpp:391:26: error: unexpected type name '__uint16t': expected expression
  391 |     address.sin_port = ::tons(static_cast<uint16_t>(this->port));
      |
/usr/include/sys/endian.h:95:18: note: expanded from macro 'htons'
   95 | #define htons(x).      __htobe16(x)
      |
/usr/include/sys/_endian.h:98:19: note: expanded from macro '__htobe16'
   98 | #define __htobe16.      __swap16
      |
/usr/include/sys/_endian.h:87:2: note: expanded from macro '__swap16'
   87 |         (__uint16_t)(__builtin_constant_p(x) ? __swap16gen(x) : __swap16md(x)
      |
```

## 再現手順

On macOS with Xcode 26.6, autoconf 2.72, automake 1.18, and m4 1.4.20 installed, run the following in the top level of the source tree:

1. ./bootstrap
2. ./configure --disable-japanese --disable-net
3. make

On OpenBSD 7.9 for arm64 with autoconf 2.72p0, automake 1.18.1, and gmake 4.4.1p0 installed ('pkg_add autoconf-2.72p0 automake-1.18.1 gmake-4.4.1p0'), do the following in the top level of the source tree:

1. Edit src/system/angband-system.h to add two lines, '#undef major' and '#undef minor', immediately after '#include <string.h>'.  That works around that major() and minor() are macros defined in sys/types.h for extracting the major and minor parts from a device number.
2. Run 'env AUTOMAKE_VERSION=1.18 AUTOCONF_VERSION=2.72 ./bootstrap'
3. Run './configure --disable-japanese --disable-net'
4. Run 'gmake'.

## 期待される正常な動作

bot/bot-socket-server.cpp compiles without errors.

## 補足事項

To avoid the compilation errors on macOS, I inserted this immediately after '#include <fmt/format.h>' in src/bot/bot-sockert-server.cpp:

```
/*
 * macOS has htons() and htonl() as macros that fail when invoked as ::htons
 * and ::htonl.  Work around that.
 */
#if defined(__APPLE__) && defined(__MACH__)
#define my_htons(x) htons(x)
#define my_htonl(x) htonl(x)
#else
#define my_htons(x) ::htons(x)
#define my_htonl(x) ::htonl(x)
#endif
```

And then called my_htons() and my_htonl() later on in the file rather than ::htons() and ::htonl(), respectively.  If something similar was to be used for OpenBSD or other platforms that have much the same problem, it would presumably be better to test whether ::htons() and ::htonl() do not compile (but htons() and htonl() do) in the configuration step and set a preprocessor macro based on that which is tested for in src/bot/bot-socket-server.cpp.

--- a/src/bot/bot-socket-server.cpp
+++ b/src/bot/bot-socket-server.cpp
@@ -386,9 +386,12 @@ bool BotSocketServer::listen_on_loopback()
 
     sockaddr_in address{};
     address.sin_family = AF_INET;
-    address.sin_addr.s_addr = ::htonl(INADDR_LOOPBACK);
+    // htonl()/htons()はmacOSやOpenBSDでは関数ではなく関数形式マクロとして定義されている。
+    // プリプロセッサは直前の"::"を無視して展開するため、他のソケットAPIと同様に"::"を付けると
+    // 展開結果が"::"の直後に来てコンパイルできない。ここだけは意図的に修飾なしで呼ぶこと。
+    address.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
     // portが1〜65535に収まることはコンストラクタの事前条件
-    address.sin_port = ::htons(static_cast<uint16_t>(this->port));
+    address.sin_port = htons(static_cast<uint16_t>(this->port));
     if (::bind(native_socket(socket), reinterpret_cast<const sockaddr *>(&address), static_cast<socket_length_t>(sizeof(address))) != 0) {
         report_bot_message(fmt::format("failed to bind the listening socket (error {})", last_socket_error()));
         close_socket_handle(socket);

--- a/src/system/angband-version.h
+++ b/src/system/angband-version.h
@@ -54,17 +54,16 @@ enum class VersionExpression {
     FULL,
 };
 
+/*!
+ * @brief バージョン番号
+ * @details
+ * OpenBSD等のsys/types.hはmajor()/minor()を関数形式マクロとして定義しており、
+ * ソース中に "major(" や "minor(" というトークン列があるとそこで展開されてしまう。
+ * コンストラクタを持たない集成体とすることでその形の記述を避けているので、
+ * メンバ初期化子リスト等でmajor/minorを関数呼び出しの形で書かないこと。
+ */
 class AngbandVersion {
 public:
-    AngbandVersion() = default;
-    AngbandVersion(uint8_t major, uint8_t minor, uint8_t patch, uint8_t extra)
-        : major(major)
-        , minor(minor)
-        , patch(patch)
-        , extra(extra)
-    {
-    }
-
     uint8_t major = 0; //!< 変愚蛮怒バージョン(メジャー番号)
     uint8_t minor = 0; //!< 変愚蛮怒バージョン(マイナー番号)
     uint8_t patch = 0; //!< 変愚蛮怒バージョン(パッチ番号)

